lib/consumerGroupMigrator.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+const assert = require('assert');
+const util = require('util');
+const async = require('async');
+const debug = require('debug')('kafka-node:ConsumerGroupMigrator');
+const zookeeper = require('node-zookeeper-client');
+const _ = require('lodash');
+const EventEmiter = require('events').EventEmitter;
+
+function ConsumerGroupMigrator (consumerGroup) {
+  assert(consumerGroup);
+  const self = this;
+  this.consumerGroup = consumerGroup;
+  this.client = consumerGroup.client;
+
+  if (consumerGroup.options.migrateRolling) {
+    this.zk = zookeeper.createClient(consumerGroup.client.connectionString);
+    this.zk.on('connected', function () {
+      self.filterByExistingZkTopics(function (topics) { // no error in callback
+        if (topics.length) {
+          self.listenForChangesToTopics(topics);
+        } else {
+          debug('No HLC topics exist in zookeeper.');
+          self.connectConsumerGroup();
+        }
+      });
+    });
+
+    this.on('noOwnersForTopics', function () {
+      debug('No owners for topics reported.');
+      self.connectConsumerGroup();
+    });
+
+    this.on('topicOwnerChange', _.debounce(function (topics) {
+      self.listenForChangesToTopics(topics);
+    }, 250));
+
+    this.zk.connect();
+  } else {
+    this.connectConsumerGroup();
+  }
+}
+
+util.inherits(ConsumerGroupMigrator, EventEmiter);
+
+ConsumerGroupMigrator.prototype.connectConsumerGroup = function () {
+  debug('%s connecting consumer group', this.client.clientId);
+  const self = this;
+  if (this.client.ready) {
+    this.consumerGroup.connect();
+  } else {
+    this.client.once('ready', function () {
+      self.consumerGroup.connect();
+    });
+  }
+  this.zk && this.zk.close();
```

Comment:
Once you join the new consumer group the migrator's job is done and there's no need to keep the zookeeper connection open.